loopd/swapclient_server.go

```diff
@@ -992,9 +1002,50 @@ func validateLoopOutRequest(chainParams *chaincfg.Params, confTarget int32,
 	}
 
 	// Check that the label is valid.
-	if err := labels.Validate(label); err != nil {
+	if err := labels.Validate(req.Label); err != nil {
 		return 0, err
 	}
 
-	return validateConfTarget(confTarget, loop.DefaultSweepConfTarget)
+	channels, err := lnd.ListChannels(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	unlimitedChannels := len(req.OutgoingChanSet) == 0
+	outgoingChanSetMap := make(map[uint64]bool)
+	for _, c := range req.OutgoingChanSet {
+		outgoingChanSetMap[c] = true
+	}
+
+	var availabeBalance btcutil.Amount
+	for _, c := range channels {
+		// Don't bother adding the local balance of an inactive channel.
+		if !c.Active {
+			continue
+		}
+
+		// If no outgoing channel set was specified then all active
+		// channel local balances are considered. However, if a channel
+		// set was specified then only local balances of the specified
+		// channel are considered.
+		if unlimitedChannels || outgoingChanSetMap[c.ChannelID] {
+			availabeBalance += c.LocalBalance
+		}
+	}
+
+	// Determine if the total outgoing capacity of the channel set is
+	// enough to satisfy the loop amount and the maximum possible
+	// routing fees that could be charged.
+	requiredBalance := btcutil.Amount(req.Amt + req.MaxSwapRoutingFee)
+	if requiredBalance > availabeBalance {
```

Comment:
Note that the balance check is slightly more complicated than this. We do MPP payments when looping out and MPP (in LND) splits equally. So really we should check splitting up to the number of channels and then see if any split satisfies the required balance condition. This will also work if only one channel has enough balance for the payment or if we split maximally.

Reply:
Also when implementing, imo it's a good idea to separate this into it's own function and then it's easier to drop once MPP splits more intelligently.

Reply:
ok cool! Didn't know this at all. Will update 👍